stdlib/Random/src/generation.jl

```diff
@@ -17,32 +17,40 @@
 ### random floats
 
 Sampler(::Type{RNG}, ::Type{T}, n::Repetition) where {RNG<:AbstractRNG,T<:AbstractFloat} =
-    Sampler(RNG, CloseOpen01(T), n)
+    Sampler(RNG, OpenOpen01(T), n)
 
 # generic random generation function which can be used by RNG implementors
 # it is not defined as a fallback rand method as this could create ambiguities
 
-function rand(r::AbstractRNG, ::SamplerTrivial{CloseOpen01{Float16}})
-    # bias correction below is equivalent to adding eps(Float16)/2, see #33222
-    z = reinterpret(Float32, (rand(r, UInt10(UInt32)) << 13)  | 0x3f800000)
+# NOTE: bias correction for OpenOpen01 samplers, see #33222
+
+function rand(r::AbstractRNG, ::SamplerTrivial{OpenOpen01{Float16}})
+    z = reinterpret(Float32, (rand(r, UInt10(UInt32)) << 13) | 0x3f800000)
```

Comment:
This can make `z` equal `prevfloat(Float32(2))`, so that `Float16(z - prevfloat(Float16(1)))` rounds to 1.

Reply:
Hm, I am surprised, in 
https://github.com/JuliaLang/julia/issues/33222#issuecomment-530780178
I checked the extrema of the new generator. Or we should really use a hard-coded constant ``Float16(reinterpret(Float32,n(UInt32(i) << 13)  | 0x3f800000) - 0.9995117f0)`` as I proposed.

Reply:
Sorry. I didn't think this fully through. You're right, it can't become 1.

Reply:
@perrutquist, thanks for bringing this up anyway, I am always uneasy operating around eps() precision, and @mschauer, thanks for checking this.